### DISCUSSION TITLE
Sprotty-theia: On clicking "Open in Diagram" from right click options in yaml file, why is diagram not opened in new tab on right side unless yaml file is edited?

### DISCUSSION THREAD

**Discussion by Samridhiseth** [eclipse-theia/theia#8610] [General] [answered]
If we open a yaml file in sprotty, and then right click on the file to select "open in Neural network Graph" option, the sprotty diagram tab doesnot open on right side in a new tab. It gives error: Referenced widget is not in the layout (Screenshot attached). But, if we edit the file once and save it and then select "open in Neural network Graph" options on right click, the diagram is opened on right side in new tab. It should open in new tab even without editing.
<img width="950" alt="error1" src="https://user-images.githubusercontent.com/30691207/95544447-bcebe080-0a18-11eb-8cad-d76dcef42ef5.PNG">
<img width="919" alt="error2" src="https://user-images.githubusercontent.com/30691207/95544449-bf4e3a80-0a18-11eb-8002-289cd1bf5ac6.PNG">

**Comment by vince-fugnitto** [comment, marked as answer]
@Samridhiseth it might be better to open the issue on the `sprotty` repo as it is a separate project. 
Would you mind sharing your setup, for example the extensions you are using for `sprotty` (is it the Theia extension or the VS Code plugin (https://github.com/eclipse/sprotty-vscode)).

Also, a quick test but can you open the `.yaml` file not as a preview (dock it in the application). My theory is it cannot find the referenced widget as a it cannot identify it as part of the layout.
